### ISSUE TITLE
🐛[BUG]: Directly constructed model wrappers silently run force autograd with create_graph=True (train-mode default)

### ISSUE THREAD

**Issue by farrisric** [NVIDIA/nvalchemi-toolkit#173]
### Version

0.2.0

### On which installation method(s) does this occur?

Pip

## Summary

A `MACEWrapper` (or any `BaseModelMixin` wrapper) constructed directly starts
in `nn.Module`'s default train mode. `MACEWrapper.forward` passes
`training=self.training` down to MACE, where it becomes
`create_graph=True, retain_graph=True` in the force autograd
(`mace/modules/utils.py::compute_forces`). Every force evaluation then builds
and retains a second-order graph — silently. Values are unaffected, so nothing
flags it; the cost is memory and backward time on every dynamics step.

`MACEWrapper.from_checkpoint` calls `wrapper.eval()` before returning, so the
alias path is safe — but `from_checkpoint` cannot open local `.model` paths,
so local checkpoints push users toward direct construction, which is exactly
the path that inherits train mode.

## Repro

```python
import torch
from nvalchemi.models.mace import MACEWrapper

raw = torch.load('my_finetuned.model', map_location='cuda', weights_only=False)
w = MACEWrapper(raw)
print(w.training)          # True
out = w(batch)             # forces computed with create_graph=True
print(out['forces'].grad_fn)  # not None
```

In our code this ran unnoticed for months of production GCMC (a downstream
`torch.compile` crash — "compiled with non-empty donated buffers" — turned out
to be a symptom: the create_graph backward needed the buffers the compiled
backward donates).

## Suggestions (any of)

- Call `self.eval()` at the end of wrapper `__init__` (training is opt-in for
  a wrapper meant for inference/dynamics), or
- warn once in `forward` when `self.training` is True and no autograd output
  requires a training graph, or
- at minimum, a prominent note in the model-wrapping docs.

Environment: nvalchemi-toolkit 0.2.0, mace-torch 0.3.x, torch 2.12.

### Minimum reproducible example

```python

```

### Relevant log output

```shell

```

### Environment details

```shell

```
